src/components/Maps/GoogleMapMarker.vue

```diff
@@ -13,37 +13,91 @@ export default {
       type: Object,
       required: true
     },
+    geocode: {
+      type: Boolean,
+      required: false
+    },
+    icon: {
+      type: String,
+      required: false
+    },
     collection: {
       type: Array,
       required: true
     }
   },
   data() {
     return {
-      theinfo: null
+      theinfo: null,
+      mapMarker: null
     };
   },
+  beforeDestroy() {
+    //console.log('remove from map', this.marker.position);
+    try {
+      this.mapMarker.setMap(null);
+    } catch (error) {
+      error;
+    }
+  },
   mounted() {
-    const { Marker, InfoWindow } = this.google.maps;
+    const { Geocoder, GeocoderStatus, Marker, InfoWindow } = this.google.maps;
 
-    var mapMarker = new Marker({
-      position: this.marker.position,
-      title: this.marker.title,
-      marker: this.marker,
-      map: this.map
-    });
+    new Promise((resolve, reject) => {
+      // If there is already a lat/lng then we are done
+      if (this.marker.position.lat && this.marker.position.lng) {
+        /* eslint no-console: 0 */
```

Comment:
eslint but console log removed

Reply:
missed that one. Ty! 7493b3d9bd9be717d44319681ff2190d165b3